sql/v2/errors/errors.go

```diff
@@ -0,0 +1,151 @@
+/*
+ Copyright 2024 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package errors
+
+import (
+	"fmt"
+	"strings"
+)
+
+type errorKind int
+
+const (
+	parseError = iota
+	mathError
+	castError
+	missingAttributeError
+	missingFunctionError
+	functionEvaluationError
+)
+
+type cesqlError struct {
+	kind    errorKind
+	message string
+}
+
+func (eKind errorKind) String() string {
+	switch eKind {
+	case parseError:
+		return "parse error"
+	case mathError:
+		return "math error"
+	case castError:
+		return "cast error"
+	case missingAttributeError:
+		return "missing attribute error"
+	case missingFunctionError:
+		return "missing function error"
+	case functionEvaluationError:
+		return "function evaluation error"
+	default:
+		return "generic error"
+	}
+
+}
+
+func (cerr cesqlError) Error() string {
+	return fmt.Sprintf("%s: %s", cerr.kind, cerr.message)
+}
+
+func NewParseError(errs []error) error {
+	if len(errs) == 0 {
+		return nil
+	}
+
+	errorMessages := make([]string, 0, len(errs))
+	for _, err := range errs {
+		errorMessages = append(errorMessages, err.Error())
+	}
+
+	return cesqlError{
+		kind:    parseError,
+		message: strings.Join(errorMessages, ","),
```

Comment:
Bold assumption that "," will never be used in an error message.... ever. I would have thought that "\n" would be better (or a less likely character, like "|", if you still wanted a single line) so that people could more easily split this into individual errors later on if needed. Not a show-stopper though.